### PR TITLE
Move ssh state tests to pytest

### PR DESCRIPTION
**Pull request by krionbsd**
### What does this PR do?
Move ssh state tests to pytest

### CODE REVIEW

File: tests/pytests/unit/states/test_ssh_known_hosts.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    name = "github.com"
+    user = "root"
+    key = "16:27:ac:a5:76:28:2d:36:63:1b:56:4d:eb:df:a6:48"
+    fingerprint = [key]
+
+    ret = {"name": name, "changes": {}, "result": False, "comment": ""}
+
+    with patch.dict(ssh_known_hosts.__opts__, {"test": True}):
+        with patch.object(os.path, "isabs", MagicMock(return_value=False)):
+            comt = 'If not specifying a "user", ' 'specify an absolute "config".'
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_ssh_known_hosts.py:30]:
```suggestion
            comt = 'If not specifying a "user", specify an absolute "config".'
```

File: tests/pytests/unit/states/test_ssh_known_hosts.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        ret.update({"comment": comt})
+        assert ssh_known_hosts.present(name, user, key=key) == ret
+
+        mock = MagicMock(side_effect=["exists", "add", "update"])
+        with patch.dict(ssh_known_hosts.__salt__, {"ssh.check_known_host": mock}):
+            comt = "Host github.com is already in .ssh/known_hosts"
+            ret.update({"comment": comt, "result": True})
+            assert ssh_known_hosts.present(name, user) == ret
+
+            comt = "Key for github.com is set to be" " added to .ssh/known_hosts"
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_ssh_known_hosts.py:48]:
```suggestion
            comt = "Key for github.com is set to be added to .ssh/known_hosts"
```

File: tests/pytests/unit/states/test_ssh_known_hosts.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            ret.update({"comment": comt, "result": None})
+            assert ssh_known_hosts.present(name, user) == ret
+
+            comt = "Key for github.com is set to be " "updated in .ssh/known_hosts"
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_ssh_known_hosts.py:52]:
```suggestion
            comt = "Key for github.com is set to be updated in .ssh/known_hosts"
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/states/test_ssh_known_hosts.py (hunk 3 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_ssh_known_hosts.py]: Fixed

File: tests/pytests/unit/states/test_ssh_auth.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+    ret = {"name": name, "changes": {}, "result": True, "comment": ""}
+
+    mock = MagicMock(return_value="exists")
+    mock_data = MagicMock(side_effect=["replace", "new"])
+    with patch.dict(
+        ssh_auth.__salt__, {"ssh.check_key": mock, "ssh.set_auth_key": mock_data}
+    ):
+        with patch.dict(ssh_auth.__opts__, {"test": True}):
+            comt = "The authorized host key sshkeys is already " "present for user root"
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_ssh_auth.py:32]:
```suggestion
            comt = "The authorized host key sshkeys is already present for user root"
```

File: tests/pytests/unit/states/test_ssh_auth.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+            ret.update({"comment": comt})
+            assert ssh_auth.present(name, user, source) == ret
+
+        with patch.dict(ssh_auth.__opts__, {"test": False}):
+            comt = "The authorized host key sshkeys " "for user root was updated"
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_ssh_auth.py:37]:
```suggestion
            comt = "The authorized host key sshkeys for user root was updated"
```

File: tests/pytests/unit/states/test_ssh_auth.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+            ret.update({"comment": comt, "changes": {name: "Updated"}})
+            assert ssh_auth.present(name, user, source) == ret
+
+            comt = "The authorized host key sshkeys " "for user root was added"
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_ssh_auth.py:41]:
```suggestion
            comt = "The authorized host key sshkeys for user root was added"
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
I missed these in the first review

File: tests/pytests/unit/states/test_ssh_auth.py (hunk 4 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_ssh_auth.py]: Fixed

File: tests/pytests/unit/states/test_ssh_auth.py (hunk 5 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_ssh_auth.py]: Fixed

File: tests/pytests/unit/states/test_ssh_auth.py (hunk 6 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_ssh_auth.py]: Fixed

File: tests/pytests/unit/states/test_ssh_known_hosts.py (hunk 2 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_ssh_known_hosts.py]: Fixed

**Review by s0undt3ch — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**